litt/README.md

```diff
@@ -0,0 +1,876 @@
+written to disk and available when the database is restarted, or the [value](#value) will be completely absent.
+It will never be the case that only part of the [value](#value) is written to disk.
+
+## Cache
+
+LittDB maintains an in-memory cache of [key](#key)-[value](#value) pairs. Data is stored in this cache when a value
+is first written, as well as when it is read from disk. This is not needed for correctness, but is rather a performance
+optimization. The cache is not persistent, and is lost when the database is restarted. The size of the cache is
+configurable.
+
+## Batched Writes
+
+LittDB supports batched write operations. Multiple write operations can be grouped together and passed to the database
+as a single operation. This may have positive performance implications, but is semantically equivalent to writing each
+value individually. A batch of writes is not [atomic](#atomicity) as a whole, but each individual write within the
+batch is [atomic](#atomicity). That is to say, if there is a crash after a batch of writes has been written but before
+it has been [flushed](#flushing), some of the writes in the batch may be [durable](#durability) on disk, while others
+may not be.
+
+## Durability
+
+In this context, the term "durable" is used to mean that data is stored on disk in such a way that it will not be lost
+in the event of a crash. Data that has been [flushed](#flushing) is considered durable. Data that has not been flushed
+is not considered durable. That doesn't mean that the data will be lost in the event of a crash, but rather that it
+is not guaranteed to be present after a crash.
+
+There are some limits to the strength of the durability guarantee provided by LittDB. For example, some drives buffer
+data in internal buffers before writing it to disk, and do not necessarily write data to disk immediately. LittDB is
+only as robust as the OS/hardware it is running on. This is true for any database, but it is worth mentioning here
+for the sake of completeness.
+
+## Flushing
+
+Calling `Flush()` causes all data previously written to be written [durably](#durability) to disk. A call to `Flush()`
+blocks until all data that was written prior to the call to `Flush()` has been written to disk.
+
+It is ok to never call `Flush()`. As internal buffers fill, data is written to disk automatically. However, calling
+`Flush()` can be useful in some cases, such as when you want to ensure that data is written to disk before proceeding
+with other operations.
+
+If `Flush()` is never called, data becomes durable through two mechanisms:
+
+- When a [segment](#segment) becomes full, it is made immutable and a new segment is created. As part of the process
+  of making a segment immutable, all data in the segment is fully written to disk.
+- When the database is cleanly stopped via a call to `Stop()`, all unflushed data is written to disk. `Stop()` blocks
```

Comment:
"blocks _until_ this has been completed"

Reply:
fixed